src/index.js

```diff
@@ -1,75 +1,112 @@
-						type == 'mixin'   ? is(x, 'function') && !!x.mixin :
-						type == 'factory' ? is(x, 'function') && !is(x, 'mixin') && !is(x, 'class') && x.length == 1 :
-						typeof x == type;
-	}
-	if (typeof x == 'object') {
-		if (x instanceof type) return true
-		if (type.class && x instanceof type.class) return true
-		if (type.mixin && type.mixin.classes) return type.mixin.classes.reduce((f,c) => f || is(x,c), false)
-	}
-	else if (typeof x == 'function') {
-		if (x.mixin && x.mixin.mixins.indexOf(type) !== -1) return true
-		var c = x
-		while (c !== Object) {
-			if (c === type || c === type.class) return true
-			if (type.mixin && type.mixin.classes && type.mixin.classes.indexOf(c) !== -1) return true
-			c = c.prototype.__proto__.constructor
-		}
-	}
-	return false
-}
+    if (typeof type === 'string') {
+        switch (type) {
+        case 'class': {
+            return is(x, 'function') && (
+                /^class\s/.test(x)
+                || /^.*classCallCheck\(/.test(x.replace(/^[^{]*{\s*/, '').replace(/\s*}[^}]*$/, ''))
+            );
+        }
+        case 'mixin': return is(x, 'function') && !!x.mixin;
+        case 'factory': return is(x, 'function') && !is(x, 'mixin') && !is(x, 'class') && x.length === 1;
+        // We already made sure that type contains a string, so it should contain a type or the API was misused.
+        // eslint-disable-next-line valid-typeof
+        default: return typeof x === type;
+        }
+    }
 
-function like(x, type) {
-	if (is(x, type)) return true
-	var itf = type.interface || (is(type, 'function') && getInterface(type.prototype))
-	var subject = is(x, 'function') ? x.interface || getInterface(x.prototype) : x
-	return itf && Object.keys(itf).reduce((f, k) => 
-		f && (is(itf[k], 'function') ? is(subject[k], 'function') : k in subject), true
-	)
-}
+    if (typeof x === 'object') {
+        if (x instanceof type) return true;
+        if (type.class && x instanceof type.class) return true;
+        if (type.mixin && type.mixin.classes) return type.mixin.classes.reduce((f, c) => f || is(x, c), false);
+    } else if (typeof x === 'function') {
+        if (x.mixin && x.mixin.mixins.indexOf(type) !== -1) return true;
+        let c = x;
+        while (c !== Object) {
+            if (c === type || c === type.class) return true;
+            if (type.mixin && type.mixin.classes && type.mixin.classes.indexOf(c) !== -1) return true;
+            c = Object.getPrototypeOf(c.prototype).constructor;
+        }
+    }
 
-function getInterface(proto) {
-	return getPropertyNames(proto).reduce((o,k) => {o[k] = proto[k]; return o}, {})
+    return false;
 }
 
-function getPropertyNames(proto) {
-	var results = []
-	while (proto !== Object.prototype) {
-		Object.getOwnPropertyNames(proto).reduce((arr,k) => arr.indexOf(k) === -1 ? arr.push(k) && arr : arr, results)
-		proto = proto.__proto__.constructor.prototype
-	}
-	return results
+function mix(...args) {
```

Comment:
Can we maybe put this back at the top?

In the original source code, the public API stuff was at the very top and the further down you go, the more private/implementation specific/low-level the code gets. That was actually done on purpose. That way someone casually inspecting the source does not have to wade through a whole lot of nitty gritty low level private functions before he gets to the ones he is used to using in his program.